agent/consul/server.go

```diff
@@ -421,11 +421,11 @@ func NewServer(config *Config, flat Deps) (*Server, error) {
 				srv:            s,
 				gatewayLocator: s.gatewayLocator,
 			},
-			Logger: s.logger,
+			Logger: s.loggers.Named(logging.Replication).Named(logging.FederationState),
```

Comment:
A replicator instance takes the `Logger` on the `ReplicatorConfig` and does this Named+Named expansion before using it for all internal logging.

When using the function replicator style, the function accepts a logger as an argument, and that logger is the derived one with the Named+Named prefix above (this is what config entries do).

The federation states version uses the delegate style, and so has it's logger configured separately here. Since this was just using `s.logger` before it was not getting the Named+Named prefix.